app/maptools/recordingmaptool.cpp

```diff
@@ -412,9 +420,83 @@ void RecordingMapTool::cancelGrab()
   }
 
   setState( MapToolState::View );
+  toggleSelectedVertexVisibility( -1 );
   setActiveVertex( Vertex() );
 }
 
+void RecordingMapTool::toggleSelectedVertexVisibility( int vertexIndex )
+{
+  if ( !mExistingVertices.isEmpty() )
+  {
+    // add currently selected vertex back to the mExistingVertices in order
+    // to make it visible again
+    if ( mActiveVertex.isValid() )
+    {
+      QgsMultiPoint *points = qgsgeometry_cast<QgsMultiPoint *>( mExistingVertices.get() );
+      points->addGeometry( mActiveVertex.coordinates().clone() );
+    }
+
+    if ( vertexIndex != -1 )
+    {
+      mActiveVertex = mVertices.at( vertexIndex );
+
+      // remove newly selected vertex from the mExistingVertices
+      QgsPoint p = mActiveVertex.coordinates();
+      mExistingVertices.filterVertices( [p]( const QgsPoint & point )->bool
+      {
+        return !InputUtils::equals( point, p, 1e-16 );
+      } );
+    }
+    emit existingVerticesChanged( mExistingVertices );
+  }
+}
+
+void RecordingMapTool::toggleHandleVisibility()
+{
+  double minDistance = std::numeric_limits<double>::max();
+  double currentDistance = 0;
+
+  if ( !mHandles.isEmpty() && mActiveVertex.isValid() )
+  {
+    // add previously hidden handle back
+    if ( !mHiddenHandle.isEmpty() )
+    {
+      QgsMultiLineString *lines = qgsgeometry_cast<QgsMultiLineString *>( mHandles.get() );
+      lines->addGeometry( mHiddenHandle.clone() );
+      mHiddenHandle = QgsGeometry();
+    }
+
+    // find part containing active vertex and hide it
+    QgsVertexId id;
+    QgsPoint point;
+    int part = -1;
+    const QgsAbstractGeometry *geom = mHandles.constGet();
+    while ( geom->nextVertex( id, point ) )
+    {
+      currentDistance = point.distance( mActiveVertex.coordinates() );
+      if ( currentDistance < minDistance )
+      {
+        minDistance = currentDistance;
+        part = id.part;
+      }
+    }
+
+    if ( part != -1 )
+    {
+      mmHiddenHandle.set( geom->lineStringN( part ).clone() );
+      mHandles.deletePart( part );
+    }
+  }
+
+  emit handlesChanged( mHandles );
+}
```

Comment:
I have mixed feelings about these functions. 
Wouldn't it be enough to just call the `createNodesAndHandles` function when a vertex is grabbed / released and inside the function just filter the `activeVertex` based on id? Do I miss something? 

Also `toggleSelectedVertexVisibility` misses the `emit activeVertexChanged()` signal